src/Microsoft.Performance.SDK/Processing/ITableDiscoverer.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Performance.SDK.Extensibility;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Provides an interface for providing tables that are exposed
+    ///     by <see cref="ProcessingSource"/>s.
+    /// </summary>
+    public interface ITableProvider
+    {
+        /// <summary>
+        ///     Returns a collection of tables.
+        /// </summary>
+        /// <param name="tableConfigSerializer">
+        ///     The serializer used to deserialize table configurations.
+        /// </param>
+        /// <returns>
+        ///     A collection of tables.
+        /// </returns>
+        ISet<DiscoveredTable> Discover(ISerializer tableConfigSerializer);
+    }
+
+    /// <summary>
+    ///     Provides common <see cref="ITableProvider"/> implementations.
+    /// </summary>
+    public static class TableDiscovery
```

Comment:
can/should we put this in its own file?

Reply:
Yes, the class is big enough that I should do that.